Windows tests fail due to call method being manually set to spawn.

<!--
IF SUFFICIENT INFORMATION IS NOT PROVIDED VIA THE FOLLOWING TEMPLATE THE ISSUE MIGHT BE CLOSED WITHOUT FURTHER CONSIDERATION OR INVESTIGATION
-->
> Please provide us with the following information:
> ---------------------------------------------------------------

### This issue is for a: (mark with an `x`)
```
- [X] bug report -> please search issues before submitting
- [ ] feature request
- [ ] documentation issue or request
- [ ] regression (a behavior that used to work and stopped in a new release)
```

`gunicorn.py` is being tested but returns an error. This testshow the application is called in Azure. 

### Minimal steps to reproduce
> _using FastAPI Postgres * ACA_
> python -m pytest


### Any log messages given by the failure
>
C:\hostedtoolcache\windows\Python\3.8.10\x64\lib\multiprocessing\context.py:193: in get_context
    raise ValueError('cannot find context for %r' % method) from None
E   ValueError: cannot find context for 'fork'
=========================== short test summary info

### Expected/desired behavior
> tests should pass like other os's

### OS and Version?
>  Windows Latest Version as ran by Github Action


### Versions
> python versions 3.8-3.12

### Mention any other details that might be useful

Some repose have suggested that we set our `call method` to `spawn`. that said we've intentionally set it to `fork` to deal with other issues that we've run into.

I suggest that we either check the operating system or add an option in the test call. 

> ---------------------------------------------------------------
> Thanks! We'll be in touch soon.

--- a/{{cookiecutter.__src_folder_name}}/src/fastapi/pyproject.toml
+++ b/{{cookiecutter.__src_folder_name}}/src/fastapi/pyproject.toml
@@ -3,13 +3,13 @@ name = "fastapi_app"
 version = "1.0.0"
 description = "{{cookiecutter.__project_short_description}}"
 dependencies = [
-    "fastapi==0.100.1",
-    "jinja2==3.1.2",
-    "uvicorn[standard]==0.23.2",
-    "python-multipart==0.0.6",
+    "fastapi",
+    "jinja2",
+    "uvicorn[standard]",
+    "python-multipart",
     {% if 'postgres' in cookiecutter.db_resource %}
-    "psycopg2-binary==2.9.6",
-    "sqlmodel==0.0.8",
+    "psycopg2",
+    "sqlmodel",
     {% endif %}
 ]
 

--- a/{{cookiecutter.__src_folder_name}}/src/tests/local/conftest.py
+++ b/{{cookiecutter.__src_folder_name}}/src/tests/local/conftest.py
@@ -12,6 +12,7 @@ import pathlib
 {% if cookiecutter.project_backend == "django" %}
 import os
 {% endif %}
+import sys
 
 {# third-party library imports #}
 import pytest
@@ -46,7 +47,8 @@ from flaskapp import db
 
 {% if cookiecutter.project_backend in ("flask", "fastapi") %}
 # Set start method to "fork" to avoid issues with pickling on OSes that default to "spawn"
-multiprocessing.set_start_method("fork")
+if sys.platform != "win32":
+    multiprocessing.set_start_method("fork")
 {% endif %}
 
 {% if cookiecutter.project_backend == "fastapi" %}

